src/metabase/models/notification.clj

```diff
@@ -370,6 +372,71 @@
           :send_once      false}
          instance))
 
+;; ------------------------------------------------------------------------------------------------;;
+;;                                         Permissions                                             ;;
+;; ------------------------------------------------------------------------------------------------;;
+
+(defn current-user-can-read-payload?
+  "Check if the current user can read the payload of a notification."
+  [notification]
+  (case (:payload_type notification)
+    :notification/card
+    (mi/can-read? :model/Card (-> notification :payload :card_id))
+
+    :notification/system-event
+    (mi/superuser?)
+
+    :notification/testing
+    true))
+
+(defn current-user-is-recipient?
+  "Check if the current user is a recipient of a notification."
+  [notification]
+  (->> (:handlers (t2/hydrate notification [:handlers :recipients]))
+       (mapcat :recipients)
+       (map :user_id)
+       set
+       (some #{(mi/current-user-id)})
+       boolean))
+
+(defn current-user-is-creator?
+  "Check if the current user is the creator of a notification."
+  [notification]
+  (= (:creator_id notification) (mi/current-user-id)))
+
+(defmethod mi/can-read? :model/Notification
+  ([notification]
+   ;; users can view a notification if they can view the payload or if they are the creator
```

Comment:
this makes it sound like it uses `current-user-can-read-payload?`, but its logic is not consistent with that 😕

Reply:
hmm yea, I'll remove the comment here since the log is clear enough